File: Ruby/pascal.rb
def pascal(n)
    pa = [[1], [1, 1]]
    for i in 1..n - 1
        x = [1]
        for j in 0..pa[-1].length - 1
            j == pa[-1].length - 1 ? (x.push(1)) : (x.push(pa[-1][j] + pa[-1][j + 1]))
        end
        pa.push(x)
    end
    return pa[0..n]
end

for i in pascal(5); p i; end

File: Python/pascal.py
def pascal(n, l = [[1], [1, 1]]):
    for i in range(1, n):
        x = [1]
        for j in range(len(l)): x.append(1) if j == len(l) - 1 else x.append(l[i][j] + l[i][j + 1])
        l.append(x)
    return l

def print_centered(n):
  l = []
  for i in n: l.append(str(i))
  m = len(l[-1])
  for i in l:
    print(" " * int((m - len(i))/2) + i + " " * (int((m - len(i))/2) + 1))

print_centered(pascal(5))
